src/modules/EdgeDetect/Module.js

```diff
@@ -19,42 +20,31 @@ module.exports = function edgeDetect(options, UI) {
 
     var step = this;
 
-    // Blur the image
-    const internalSequencer = ImageSequencer({ inBrowser: false, ui: false });
-    return internalSequencer.loadImage(input.src, function() {
-      internalSequencer.importJSON([{ 'name': 'blur', 'options': { blur: options.blur } }]);
-      return internalSequencer.run(function onCallback(internalOutput) {
-        require('get-pixels')(internalOutput, function(err, blurPixels) {
-          if (err) {
-            return;
-          }
-
-          // Extra Manipulation function used as an enveloper for applying gaussian blur and Convolution
-          function changePixel(r, g, b, a) {
-            return [(r + g + b) / 3, (r + g + b) / 3, (r + g + b) / 3, a];
-          }
-
-          function extraManipulation() {
-            return require('./EdgeUtils')(blurPixels, options.highThresholdRatio, options.lowThresholdRatio, options.hysteresis);
-          }
-
-          function output(image, datauri, mimetype, wasmSuccess) {
-            step.output = { src: datauri, format: mimetype, wasmSuccess, useWasm: options.useWasm };
-          }
-
-          return require('../_nomodule/PixelManipulation.js')(input, {
-            output: output,
-            ui: options.step.ui,
-            changePixel: changePixel,
-            extraManipulation: extraManipulation,
-            format: input.format,
-            image: options.image,
-            inBrowser: options.inBrowser,
-            callback: callback,
-            useWasm: options.useWasm
-          });
-        });
-      });
+    // Makes the image greyscale
+    function changePixel(r, g, b, a) {
+      return [(r + g + b) / 3, (r + g + b) / 3, (r + g + b) / 3, a];
+    }
+
+    // Extra Manipulation function used as an enveloper for applying gaussian blur and Convolution
+    function extraManipulation(pixels) {
+      const blurPixels = Blur(pixels, options.blur);
```

Comment:
This is great, but was it just optimization or are we not able to do internal sequences anymore?

Reply:
@jywarren I guess it just and optimization and it amazing as well

Reply:
Let's leave this comment open so people know!